src/datasets/utils/tf_utils.py

```diff
@@ -98,10 +107,12 @@ def np_get_batch(
             if key in cols_to_retain or key in ("label", "label_ids", "labels")
         }
 
-    actual_size = len(list(batch.values())[0])  # Get the length of one of the arrays, assume all same
-    # Our collators expect a list of dicts, not a dict of lists/arrays, so we invert
-    batch = [{key: value[i] for key, value in batch.items()} for i in range(actual_size)]
-    batch = collate_fn(batch, **collate_fn_args)
+    if is_batched:
+        actual_size = len(list(batch.values())[0])  # Get the length of one of the arrays, assume all same
+        # Our collators expect a list of dicts, not a dict of lists/arrays, so we invert
+        batch = [{key: value[i] for key, value in batch.items()} for i in range(actual_size)]
+        batch = collate_fn(batch, **collate_fn_args)
```

Comment:
We still need to call collate_fn if not `batched`
```suggestion
    batch = collate_fn(batch, **collate_fn_args)
```